Sources/CvPlot.cpp

```diff
@@ -16131,17 +15824,14 @@ bool CvPlot::isBorder(bool bIgnoreWater) const
 {
 	for (int iI = 0; iI < NUM_DIRECTION_TYPES; ++iI)
```

Comment:
This can be something like:
```cpp
if(algo::any_of(
    adjacent(), 
    (CvPlot::fn::isWater() && !bIgnoreWater && !isWater()) 
    ||
    (CvPlot::fn::getOwner() != getOwner())
)) 
{
    return true;
}
```